lib/github/hooks/issue.ts

```diff
@@ -427,6 +426,99 @@ export const onPullRequestOpened = async (webhooks: Webhooks) => {
+
+    // Early exit if the pull request was not merged
+    if (!pullRequest.merged) {
+      console.log("Pull request was not merged.");
+      return;
+    }
+
+    const octokit = getOctokitInstance(context.payload.installation?.id!);
+
+    // Parse the pull request body to find linked issues (e.g., " #123")
+    const issueNumbers = extractIssueNumbersFromPrBody(pullRequest.body!);
+    if (!issueNumbers.length) {
+      console.log("No issues are mentioned in the pull request.");
+      return;
+    }
+
+    // Check repository enrollment first
+    const ossGgRepo = await getRepositoryByGithubId(context.payload.repository.id);
+    if (!ossGgRepo) {
+      console.log("Repository is not enrolled in oss.gg.");
+      return;
+    }
+
+    // Loop through all the extracted issue numbers
+    for (const issueNumber of issueNumbers) {
+      try {
+        // Fetch the issue details using the GitHub API
+        const { data: issue } = await octokit.issues.get({
+          owner,
+          repo,
+          issue_number: issueNumber,
+        });
+
+        // Get the labels of the linked issue
+        const issueLabels = issue.labels.map((label: { name: string }) => label.name);
+
+        // Check if the "🕹️ oss.gg" label is present, skip if not
+        const ossLabel = issueLabels.find((label) => label.includes(OSS_GG_LABEL));
+        if (!ossLabel) {
+          console.log(`Issue #${issueNumber} does not have the 🕹️ oss.gg label. Skipping.`);
+          continue;  // Skip this issue and move to the next one
+        }
+
+        // Extract points from labels like "🕹️ 50 points", "🕹️ 100 points", etc.
+        const pointsLabel = issueLabels.find((label) => label.includes("🕹️") && label.includes("points"));
+        if (pointsLabel) {
+          // Extract number from the points label
+          const points = parseInt(pointsLabel.match(/\d+/)?.[0] || "0", 10);
+          console.log(`Points for issue #${issueNumber}:`, points);
+
+          // Fetch PR author details only if the oss.gg label is present
+          const prAuthorGithubId = pullRequest.user.id;
+          const prAuthorUsername = pullRequest.user.login;
+          const { data: prAuthorProfile } = await octokit.users.getByUsername({
+            username: prAuthorUsername,
+          });
+
+          // Fetch or create user profile only if the oss.gg label is present
+          let user = await getUserByGithubId(prAuthorGithubId);
+          if (!user) {
+            user = await createUser({
+              githubId: prAuthorGithubId,
+              login: prAuthorUsername,
+              email: prAuthorProfile.email,
+              name: prAuthorProfile.name,
+              avatarUrl: pullRequest.user.avatar_url,
+            });
+          }
+
+          // Award points to the user
+          await assignUserPoints(
+            user?.id,
+            points,
+            "Awarded points",
+            pullRequest.html_url,
+            ossGgRepo?.id
+          );
```

Comment:
after the points are awarded, we want to add the same comment to the PR we issue on onAwardPoints. Here, also make sure to follow the DRY coding principle.